packages/devextreme/js/__internal/grids/grid_core/keyboard_navigation/m_keyboard_navigation.ts

```diff
@@ -1733,17 +1733,19 @@ export class KeyboardNavigationController extends modules.ViewController {
     return this._isCellValid($cell);
   }
 
-  _isLastRow(rowIndex) {
-    const dataController = this._dataController as any;
-    const visibleItems = dataController
-      .items()
-      .filter((item) => item.visible !== false);
+  private _isLastRow(rowIndex: number): boolean {
```

Comment:
Minor comment.

Let's get rid of the local ```dataController``` variable
Something like this:
```typescript
    if (this._isVirtualRowRender()) {
      return rowIndex >= (this._dataController as any).getMaxRowIndex();
    }
    
    const itemVisibleIndexes = this._dataController.items()
      .map((item, index) => (item.visible !== false ? index : -1));
    const lastVisibleIndex = Math.max(itemVisibleIndexes);
    
    return rowIndex === lastVisibleIndex;
```

Reply:
Don't want to lose 3 approves + it's not my change, it was so before. I'll keep it in mind for next PRs